File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> answer = new ArrayList<>(); 
        Map<String, List<String>> hashMap = new HashMap<>(); 
        for(String s : strs){
            char[] toChar = s.toCharArray(); 
            Arrays.sort(toChar); 
            String key = new String(toChar); 

            hashMap.computeIfAbsent(key, k -> new ArrayList<>()).add(s); 
        }

        for(var e : hashMap.entrySet()){
            answer.add(e.getValue()); 
        }
        return answer; 
    }
}

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> answer; 
        map<string, vector<string>> hashMap; 

        for(string& s : strs){
            string tmp = s; 
            sort(tmp.begin(), tmp.end()); 

            hashMap[tmp].push_back(s); 
        }

        for(auto& it : hashMap){
            answer.push_back(it.second); 
        }

        return answer; 
    }
};